### PR TITLE
ILL SANS data reduction: read parameters from Nexus

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

During SANS data reduction (ILL), some parameters are taken from the nexus file fields:

* *sample thickness.* If the user set `SampleThickness` to -1, the algorithm will try to get it from the nexus file. If the value does not exist, the default value (0.1cm) will be used.

* *output workspace title.* When processing the sample, the output workspace will get its title from the nexus file.

**To test:**

* With data from `ExternalData/Testing/Data/UnitTest/ILL/D33/`
```python
SANSILLAutoProcess(
    SampleRuns='001464',
    SampleThickness=-1,
    OutputWorkspace='ws1'
    )
```
Logs a warning saying that the sample thickness is not in the sample logs and uses 0.1cm instead. The output workspace title is "1_F1 anethol".

* With data from `ExternalData/Testing/Data/DocTest/ILL/D11/`
```python
SANSILLAutoProcess(                                                 
    SampleRuns='2889',
    SampleThickness=-1,
    OutputWorkspace="ws2"
)
```
Does not log any warning about the sample thickness. The output workspace title is "H2O_1".

Part of #26906 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py [hunk 1]
```diff
@@ -545,6 +547,46 @@ def _apply_masks(self, ws):
...
+            Perform the normalization by sample thickness. In case the provided
+            thickness is -1, this method will try to get it from the sample
+            logs.
+            @param ws : input workspace on wich the normalization is applied.
+        """
+        thickness = self.getProperty('SampleThickness').value
+        if thickness == -1:
+            try:
+                run = mtd[ws].getRun()
+                thickness = run.getLogData('sample.thickness').value
```
Comment (gvardany) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py:561]: When using the value from the nexus file, can you log an information what that value is?

**Review by gvardany — COMMENTED** [review]
Works fine, can you make it clear in the documentation about the convention of -1 ?
Can you add a test in SANSILLReductionTest to cover this?

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py (hunk 1 above)
Comment (gui-co) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py:562]: Done

**Comment by gui-co** [comment]
I changed the doc of the property to explain the -1 special value.

Concerning the test, I just added `SampleThickness=-1` to check that the value is valid. How can I check which value has been used ? Do you want me to check that ?

**Comment by gvardany** [comment]
That would be nice. You should be able to get what value is used from the history of the output workspace. In any case, do not modify the existing test, add a new one with -1. The rest is OK.

**Comment by gui-co** [comment]
There is now a specific test that check the sample thickness value.

**Review by gvardany — APPROVED** [review]
Looks good.

**Pull request merged**